test/unit/lib/urlutilTest.js

```diff
@@ -208,6 +208,16 @@ describe('urlutil', function () {
     })
   })
 
+  describe('getPDFViewerUrl', function () {
+    const baseUrl = 'chrome-extension://jdbefljfgobbmcidnmpjamcbhnbphjnb/content/web/viewer.html?file='
+    it('regular url', function () {
+      assert.equal(urlUtil.getPDFViewerUrl('http://example.com'), baseUrl + 'http%3A%2F%2Fexample.com')
+    })
+    it('file url', function () {
+      assert.equal(urlUtil.getPDFViewerUrl('file:///Users/yan/some files/test.pdf'), baseUrl + 'file%3A%2F%2F%2FUsers%2Fyan%2Fsome%20files%2Ftest.pdf')
```

Comment:
Possibly a silly question, but will these work on other machines? Seems to me the path is hardcoded.

Reply:
@pureooze since the function just builds the path (using the provided input), the actual input itself isn't important 😄  The important part to note is that the output is prefixed with `baseUrl` (and the parts afterword are URL encoded)